src/wasm-interpreter.h

```diff
@@ -4694,12 +4712,24 @@ class ModuleRunnerBase : public ExpressionRunner<SubType> {
 #if WASM_INTERPRETER_DEBUG
     std::cout << self()->indent() << "resuming func " << func << '\n';
 #endif
-    Flow ret = callFunction(func, arguments);
+    Flow ret;
+    {
+      // Create a stack value scope. This value will never be suspended - it is
+      // the value flowing out - but it is simpler to add this here, than to
+      // special-case in visit() the toplevel expression (i.e., that has no
+      // parent expression to add a stack value scope for it).
```

Comment:
I don't quite understand 1) what this is for or 2) why special-casing the toplevel expression would be an alternative to this. (I assume understanding (1) will make (2) clear.)

Reply:
If we don't special-case here, then there is no value scope at the top level. Without a value scope, we'd error when we try to add a value to the scope. So we would need to do "if scope exists" in some places. This avoids that.

Reply:
I tried to clarify the comment now, PTAL